pages/api/correiosres/v1/index.js

```diff
@@ -0,0 +1,148 @@
+};
+
+// Padrões de alertas dos Correios definidos por expressões regulares.
+const padroesAlertasCorreios = [
+    {
+        regex: /alert\(\'CEP de origem n.*o encontrado na base de dados dos Correios \(-1\).\'\)/,
+        mensagem: ERROR_MESSAGES['CEP_ORIGEM_NOT_FOUND']
+    },
+    {
+        regex: /alert\(\'CEP de destino n.*o encontrado na base de dados dos Correios \(-2\).\'\)/,
+        mensagem: ERROR_MESSAGES['CEP_DESTINO_NOT_FOUND']
+    },
+    {
+        regex: /alert\(\'CEP de destino inv.*lido.\'\)/,
+        mensagem: ERROR_MESSAGES['CEP_DESTINO_INVALIDO']
+    },
+    {
+        regex: /alert\(\'CEP inv.*lido.\'\)/,
+        mensagem: ERROR_MESSAGES['CEP_ORIGEM_INVALIDO']
+    }
+];
+
+// Função para verificar se existe algum alerta na resposta dos Correios.
+function verificarAlertasCorreios(correiosData) {
+    const padrao = padroesAlertasCorreios.find(p => p.regex.test(correiosData));
+    return padrao ? { alerta: true, mensagem: padrao.mensagem } : { alerta: false, mensagem: '' };
+}
+
+// Função para validar a presença dos CEPs de origem e destino.
+function validarCEP(cepOrigem, cepDestino) {
+    if (!cepOrigem || !cepDestino) {
+        throw new NotFoundError({
+            message: !cepOrigem ? ERROR_MESSAGES['CEP_ORIGEM_VAZIO'] : ERROR_MESSAGES['CEP_DESTINO_VAZIO'],
+            type: 'cep_error',
+            name: 'CEP_NOT_FOUND',
+        });
+    }
+}
+
+// Função para analisar o HTML retornado pelos Correios e extrair informações.
+function parseCorreiosHtml(correiosData) {
+    const dom = new jsdom.JSDOM(correiosData);
+    const errorDiv = dom.window.document.querySelector('div[class="info error"]');
+    const positiveDiv = dom.window.document.querySelector('div[class="informativo-positivo"]');
+
+    return { errorDiv, positiveDiv };
+}
+
+// Função principal que verifica restrições de entrega para um CEP.
+const verifyRestriction = async (request, response) => {
+    try {
+        const {cepOrigem, cepDestino, servico} = request.query;
+
+        // Verifica se o serviço solicitado é válido.
+        if (!(servico in SERVICOS)) {
+            throw new InternalError({
+                message: ERROR_MESSAGES['ERRO_SERVICO'],
+                type: 'internal_error',
+                name: 'INTERNAL_ERROR',
+            });
+        }
+
+        // Valida os CEPs fornecidos na requisição.
+        validarCEP(cepOrigem, cepDestino);
+        
+        // Solicita os dados dos Correios e verifica por alertas.
+        const correiosData = await requestHtmlCorreios(cepOrigem, cepDestino, SERVICOS[servico]);
+        const verificarAlerta = verificarAlertasCorreios(correiosData);
+
+        // Se um alerta for encontrado, lança um erro.
+        if (verificarAlerta['alerta']) {
+            throw new NotFoundError({
+                message: verificarAlerta['mensagem'],
+                type: 'cep_error',
+                name: 'CEP_NOT_FOUND',
+            });
+        }
```

Comment:
Partindo do comentário https://github.com/BrasilAPI/BrasilAPI/pull/557/files#r1412902104

A gnt poderia reescrever esse trecho pra algo tipo

```suggestion
        const foundAlert = getAlert(correiosData);

        if (foundAlert) {
            throw new NotFoundError({
                message: foundAlert,
                type: 'cep_error',
                name: 'CEP_NOT_FOUND',
            });
        }
```